lib/couchbase-orm/utilities/query_helper.rb

```diff
@@ -18,6 +18,8 @@ def build_match(key, value)
                     "(#{build_match(key, nil)} OR #{build_match(key, value.compact)})"
                 when value.is_a?(Array)
                     "#{key} IN #{quote(value)}"
+                when value.instance_of?(Range)
```

Comment:
I'd rather also accept any instance which class inherit from Range... or maybe any object responding to `:begin` and `:end`

```suggestion
                when value.is_a?(Range)
```

![image](https://github.com/doctolib/couchbase-orm/assets/462084/c7c1e39d-4c7d-4794-b344-4c24909f94cd)